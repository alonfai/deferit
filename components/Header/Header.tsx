import React from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { Foundation } from '@expo/vector-icons';
import { helpers } from '../../utils';

export type Props = {};

const device = helpers.getDevice();

const Header: React.FC<Props> = () => {
  return (
    <View style={styles.container}>
      <Foundation name='dollar-bill' size={24} color='black' />
      <Text style={styles.title}>Bill List</Text>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    height: Dimensions.get('window').height / 10,
    alignItems: 'center',
    paddingTop: device === 'Mobile' ? 12 : 6,
    justifyContent: 'center',
    flexDirection: 'row',
    backgroundColor: 'royalblue',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginLeft: 10,
  },
});
export default Header;
