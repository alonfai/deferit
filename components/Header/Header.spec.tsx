import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';

import Header from './Header';

describe('<Header />', () => {
  it('general', () => {
    const allQuestions = ['q1', 'q2'];
    const mockFn = jest.fn();

    const { getAllByA11yLabel, getByText } = render(
      // <Header questions={allQuestions} onSubmit={mockFn} />
      <Header />
    );
    const answerInputs = getAllByA11yLabel('answer input');

    fireEvent.changeText(answerInputs[0], 'a1');
    fireEvent.changeText(answerInputs[1], 'a2');
    fireEvent.press(getByText('Submit'));

    expect(mockFn).toBeCalledWith({
      '1': { q: 'q1', a: 'a1' },
      '2': { q: 'q2', a: 'a2' },
    });
  });
});
