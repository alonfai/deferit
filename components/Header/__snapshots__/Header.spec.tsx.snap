// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Render successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "royalblue",
      "flexDirection": "row",
      "height": 133.4,
      "justifyContent": "center",
      "paddingTop": 12,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginLeft": 10,
      }
    }
  >
    Bill List
  </Text>
</View>
`;
